In [5]:
import requests
from bs4 import BeautifulSoup
from os.path import basename, splitext


In [6]:
proxies = {
  "https": "http://127.0.0.1:1080",
}

#Note please remove the proxy, as you can connect Google directly
r = requests.get('https://www.google.com.hk/search?safe=on&q=hello+kitty&spell=3&sa=X', proxies=proxies)

In [7]:
print(r.text)

<!doctype html><html itemscope="" itemtype="http://schema.org/SearchResultsPage" lang="en-GB"><head><meta content="text/html; charset=UTF-8" http-equiv="Content-Type"><meta content="/images/branding/googleg/1x/googleg_standard_color_128dp.png" itemprop="image"><link href="/images/branding/product/ico/googleg_lodp.ico" rel="shortcut icon"><title>hello kitty - Google Search</title><style>#gbar,#guser{font-size:13px;padding-top:1px !important;}#gbar{height:22px}#guser{padding-bottom:7px !important;text-align:right}.gbh,.gbd{border-top:1px solid #c9d7f1;font-size:1px}.gbh{height:0;position:absolute;top:24px;width:100%}@media all{.gb1{height:22px;margin-right:.5em;vertical-align:top}#gbar{float:left}}a.gb1,a.gb4{text-decoration:underline !important}a.gb1,a.gb4{color:#00c !important}.gbi .gb4{color:#dd8e27 !important}.gbf .gb4{color:#900 !important} </style><style>.star{float:left;margin-top:1px;overflow:hidden}.ybhkme{font-size:11px}.j{width:34em}body,td,div,.p,a{font-family:arial,sans-seri

In [8]:
soup = BeautifulSoup(r.text,'html.parser')
for img in soup.findAll('img'):
    print(img['src'])


https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTkdHXloFhAzUsQu7MT5E0vMiAzrRL1e2qIrOWG0UixxFlk6AiThZ9OGM8
https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRF1UU_1NoNEDX36JfYorZGwfND2z65RqaPPdIp7Z3fC7MSR5k3vG7RvA
https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTdJws1iv1IXD7eSU4TFwQ2XzpBLBQRiY9qm0XMJihnZY-BbXmlCM_fvw
https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSV9iJ9Q7MlJc57LHT85cROxnLuDQWT5iI_I5FMT52F4I0-lsPOulU6RA
https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQzZPFeE22DwupbqbKdjbI6uWkoO7a1AWlepY007ddtisZTJeQIe9m48so
https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRPHZHrbPjX0sEgQ3Ks-hMNm42pB9zTr6LDfaol9F7x3v17z10hG8Hyg-4
https://img.youtube.com/vi/mtw68bn61qM/default.jpg?h=90&w=120&sigh=__O1OBY87d1YK3Twekm160JIYmIrQ=
https://img.youtube.com/vi/30PXEtSX14U/default.jpg?h=90&w=120&sigh=__LgGDTnOz3_DbG7vhUk2RCDrmGbA=
https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSOyA-ioudkRPHkJ3jxzIH7NI_kBGsLm-MNzvG9mYTvpOTLS317ktTsPw
https://encrypted-tbn1.g

# Requests&Get

## requests then download and save to ./images

In [10]:
from bs4 import BeautifulSoup
import requests
import re
import os
import argparse
import sys
import json
from PIL import Image
from io import BytesIO


In [11]:
query='hello+kitty'
url="https://www.google.com.hk/search?q="+query+"&source=lnms&tbm=isch"
header={'User-Agent':"Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.134 Safari/537.36"}

r = requests.get(url,headers=header, proxies=proxies)


soup = BeautifulSoup(r.text,'html.parser')
ActualImages=[]# contains the link for Large original images, type of  image
for a in soup.find_all("div",{"class":"rg_meta"}):
    link , Type =json.loads(a.text)["ou"]  ,json.loads(a.text)["ity"]
    ActualImages.append((link,Type))
    print(link)

https://sanrio-production-weblinc.netdna-ssl.com/media/W1siZiIsIjIwMTYvMDYvMTQvMjAvNDgvMzQvMTM3L2NocmFjdGVyX2hlbGxvX2tpdHR5LmpwZyJdXQ/chracter-hello-kitty.jpg?sha=f5e7c272d3fc6e78
https://upload.wikimedia.org/wikipedia/en/thumb/0/05/Hello_kitty_character_portrait.png/200px-Hello_kitty_character_portrait.png
https://i.guim.co.uk/img/static/sys-images/Guardian/Pix/pictures/2014/8/28/1409228903473/de6aa8da-3f82-4069-9807-0e83513dad05-1024x768.jpeg?w=300&q=55&auto=format&usm=12&fit=max&s=f74cf7108fab04cf08066db9225d52c2
https://vignette.wikia.nocookie.net/epic-rap-battles-of-cartoons/images/1/10/Hello-kitty.png/revision/latest?cb=20170807065310
https://pbs.twimg.com/profile_images/553014757037322240/VhIzHhkV_400x400.png
https://seeklogo.com/images/H/hello-kitty-logo-354D6628E7-seeklogo.com.png
https://www.sanrio.com/media/W1siZiIsIjIwMTYvMDYvMTMvMTQvMTEvNDAvMTQvY2hhcmFjdGVyX2Jhbm5lcl9oZWxsb2tpdHR5LnBuZyJdXQ/character_banner_hellokitty.png?sha=95006e8644727395
https://sophosnews.files.wordp

In [12]:
for i , (img , Type) in enumerate(ActualImages[0:100000]):
    try:
        req = requests.get(img, headers=header, proxies=proxies)
        im = Image.open(BytesIO(req.content))
        rgb_im = im.convert('RGB')
        
        rgb_im.save(os.path.join("./images/" , "img" + "_"+ str(i)+".jpg"))
        print("saved",i)
    except Exception as e:
        print ("could not load : "+img)
        print(e)

saved 0
saved 1
saved 2
saved 3
saved 4
saved 5
saved 6
saved 7
saved 8
saved 9
saved 10
could not load : http://bdfjade.com/data/out/1/5108168-hello-kitty-images.jpg
cannot identify image file <_io.BytesIO object at 0x000001CD70536360>
saved 12
saved 13
saved 14
saved 15
saved 16
saved 17


C:\Users\forqzy\AppData\Local\conda\conda\envs\tensorflow\lib\site-packages\PIL\Image.py:888: UserWarning: Palette images with Transparency   expressed in bytes should be converted to RGBA images
  'to RGBA images')


saved 18
saved 19
saved 20
saved 21
saved 22
saved 23
saved 24
saved 25
saved 26
saved 27
saved 28
saved 29
saved 30
saved 31
saved 32
saved 33
saved 34
saved 35
saved 36
saved 37
saved 38
saved 39
saved 40
saved 41
saved 42
saved 43
saved 44
saved 45
saved 46
saved 47
could not load : http://bdfjade.com/data/out/26/5349338-wallpaper-hello-kitty.jpg
cannot identify image file <_io.BytesIO object at 0x000001CD70536410>
saved 49
saved 50
saved 51
saved 52
saved 53
saved 54
saved 55
saved 56
saved 57
saved 58
saved 59
saved 60
saved 61
saved 62
saved 63
saved 64
saved 65
saved 66
saved 67
saved 68
saved 69
could not load : http://4.bp.blogspot.com/-S4OtVKoCX_g/UpUEwAMGDSI/AAAAAAABTtc/-H00vlWxOTM/s1600/Hello+Kitty+69.png
HTTPConnectionPool(host='4.bp.blogspot.com', port=80): Max retries exceeded with url: /-S4OtVKoCX_g/UpUEwAMGDSI/AAAAAAABTtc/-H00vlWxOTM/s1600/Hello+Kitty+69.png (Caused by NewConnectionError('<requests.packages.urllib3.connection.HTTPConnection object at 0x000001CD705EFE10

# Homework
  Resize the image to 160*120 using PIL.Image
  
  Hint' Copy the above code and add a line of code to resize the image then save to reduce the time needed. 
    
    
    